viam-cartographer/src/carto_facade/carto_facade_test.cc

```diff
@@ -569,6 +569,65 @@ BOOST_AUTO_TEST_CASE(CartoFacade_config) {
     BOOST_TEST(viam_carto_lib_terminate(&lib) == VIAM_CARTO_SUCCESS);
 }
 
+BOOST_AUTO_TEST_CASE(CartoFacade_start_stop) {
+    //  validate invalid pointer
+    BOOST_TEST(viam_carto_start(nullptr) == VIAM_CARTO_VC_INVALID);
+    BOOST_TEST(viam_carto_stop(nullptr) == VIAM_CARTO_VC_INVALID);
+
+    // library init
+    viam_carto_lib *lib;
+    BOOST_TEST(viam_carto_lib_init(&lib, 0, 1) == VIAM_CARTO_SUCCESS);
+
+    // Setup
+    viam_carto *vc;
+    std::vector<std::string> sensors_vec;
+    sensors_vec.push_back("sensor_1");
+    sensors_vec.push_back("sensor_2");
+    sensors_vec.push_back("sensor_3");
+    sensors_vec.push_back("sensor_4");
+    sensors_vec.push_back("sensor_5");
+    fs::path tmp_dir =
+        fs::temp_directory_path() / fs::path(bfs::unique_path().string());
+    struct viam_carto_config vcc =
+        viam_carto_config_setup(1, VIAM_CARTO_THREE_D, tmp_dir.string(),
+
+                                "some component refereance", sensors_vec);
+    struct viam_carto_algo_config ac = viam_carto_algo_config_setup();
+
+    BOOST_TEST(viam_carto_init(&vc, lib, vcc, ac) == VIAM_CARTO_SUCCESS);
+    viam::carto_facade::CartoFacade *cf =
+        static_cast<viam::carto_facade::CartoFacade *>(vc->carto_obj);
+    BOOST_TEST(cf->started == false);
+
+    BOOST_TEST(fs::is_directory(cf->path_to_internal_state));
+    BOOST_TEST(fs::is_empty(cf->path_to_internal_state));
+
+    // // Start
+    BOOST_TEST(viam_carto_start(vc) == VIAM_CARTO_SUCCESS);
+    BOOST_TEST(cf->started == true);
+
+    // Confirm at least one map is persisted within the map_rate_sec
+    VLOG(1) << "path_to_internal_state: " << cf->path_to_internal_state;
+    // TODO: This should busy wait until this condition happens & the tests
+    // passes or or 2 seconds goes by & the test fails.
+    std::this_thread::sleep_for(cf->config.map_rate_sec +
+                                std::chrono::seconds(1));
```

Comment:
Could this cause a flaky test?

Reply:
discussed offline: could be a flaky test but that would indicate an implementation error or that the server is ridiculously over taxed